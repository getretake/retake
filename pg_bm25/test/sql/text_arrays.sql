CREATE TABLE example_table (
    id SERIAL PRIMARY KEY,
    text_array TEXT[],
    varchar_array VARCHAR[]
);

INSERT INTO example_table (text_array, varchar_array) VALUES 
('{"text1", "text2", "text3"}', '{"vtext1", "vtext2"}'),
('{"another", "array", "of", "texts"}', '{"vtext3", "vtext4", "vtext5"}'),
('{"single element"}', '{"single varchar element"}');

CREATE INDEX ON example_table
USING bm25 ((example_table.*))
WITH (key_field='id', text_fields='{text_array: {}, varchar_array: {}}');

SELECT * FROM example_table WHERE example_table @@@ 'text_array:text1';
SELECT * FROM example_table WHERE example_table @@@ 'text_array:"single element"';
SELECT * FROM example_table WHERE example_table @@@ 'varchar_array:varchar OR text_array:array';
