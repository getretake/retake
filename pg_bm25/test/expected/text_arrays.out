CREATE TABLE example_table (
    id SERIAL PRIMARY KEY,
    text_array TEXT[]
);
INSERT INTO example_table (text_array) VALUES 
('{"text1", "text2", "text3"}'),
('{"another", "array", "of", "texts"}'),
('{"single element"}');
CREATE INDEX ON example_table
USING bm25 ((example_table.*))
WITH (text_fields='{text_array: {}}');
SELECT * FROM example_table WHERE example_table @@@ 'text_array:text1';
 id |     text_array      
----+---------------------
  1 | {text1,text2,text3}
(1 row)

SELECT * FROM example_table WHERE example_table @@@ 'text_array:"single element"';
 id |     text_array     
----+--------------------
  3 | {"single element"}
(1 row)

